torchvision/models/video/s3d.py

```diff
@@ -0,0 +1,173 @@
+        x (Tensor): batch of videos with dimensions (batch, channel, time, height, width)
+    """
+
+    def __init__(
+        self,
+        num_classes: int = 400,
+    ) -> None:
+        super().__init__()
+        self.base = nn.Sequential(
+            SepConv3d(3, 64, kernel_size=7, stride=2, padding=3),
+            nn.MaxPool3d(kernel_size=(1, 3, 3), stride=(1, 2, 2), padding=(0, 1, 1)),
+            Conv3dNormActivation(
+                64, 64, kernel_size=1, stride=1, norm_layer=partial(nn.BatchNorm3d, eps=1e-3, momentum=1e-3)
+            ),
+            SepConv3d(64, 192, kernel_size=3, stride=1, padding=1),
+            nn.MaxPool3d(kernel_size=(1, 3, 3), stride=(1, 2, 2), padding=(0, 1, 1)),
+            SepInceptionBlock3D(192, [[64], [96, 128], [16, 32], [32]]),
+            SepInceptionBlock3D(256, [[128], [128, 192], [32, 96], [64]]),
+            nn.MaxPool3d(kernel_size=(3, 3, 3), stride=(2, 2, 2), padding=(1, 1, 1)),
+            SepInceptionBlock3D(480, [[192], [96, 208], [16, 48], [64]]),
+            SepInceptionBlock3D(512, [[160], [112, 224], [24, 64], [64]]),
+            SepInceptionBlock3D(512, [[128], [128, 256], [24, 64], [64]]),
+            SepInceptionBlock3D(512, [[112], [144, 288], [32, 64], [64]]),
+            SepInceptionBlock3D(528, [[256], [160, 320], [32, 128], [128]]),
+            nn.MaxPool3d(kernel_size=(2, 2, 2), stride=(2, 2, 2), padding=(0, 0, 0)),
+            SepInceptionBlock3D(832, [[256], [160, 320], [32, 128], [128]]),
+            SepInceptionBlock3D(832, [[384], [192, 384], [48, 128], [128]]),
+        )
+        self.fc = nn.Conv3d(1024, num_classes, kernel_size=1, stride=1, bias=True)
+
+    def forward(self, x):
+        y = self.base(x)
+        y = F.avg_pool3d(y, (2, y.size(3), y.size(4)), stride=1)
+        y = self.fc(y)
+        y = y.view(y.size(0), y.size(1), y.size(2))
+        logits = torch.mean(y, 2)
+
+        return logits
+
+
+class S3D_Weights(WeightsEnum):
+    pass
+
+
+@register_model()
+def s3d(*, weights: Optional[S3D_Weights] = None, progress: bool = True, **kwargs: Any) -> S3D:
+    """Construct Separable 3D CNN model.
+
+    Reference: `Rethinking Spatiotemporal Feature Learning <https://arxiv.org/abs/1712.04851>`__.
+
+    Args:
+        weights (:class:`~torchvision.models.video.S3D_Weights`, optional): The
+            pretrained weights to use. See
+            :class:`~torchvision.models.video.S3D_Weights`
+            below for more details, and possible values. By default, no
+            pre-trained weights are used.
+        progress (bool): If True, displays a progress bar of the download to stderr. Default is True.
+        **kwargs: parameters passed to the ``torchvision.models.video.S3D`` base class.
+            Please refer to the `source code
+            <https://github.com/pytorch/vision/blob/main/torchvision/models/video/s3d.py>`_
+            for more details about this class.
+
+    """
+    weights = S3D_Weights.verify(weights)
+
```

Comment:
You need the following bit. In case someone asks for a pretrained model, the num_classes is predetermined.
```suggestion

    if weights is not None:
        _ovewrite_named_param(kwargs, "num_classes", len(weights.meta["categories"]))
```